src/pages/dashboardOverview/components/notActiveProductsSection/components/productFilters.ts

```diff
@@ -0,0 +1,65 @@
+import { OnboardedProduct } from '../../../../../api/generated/b4f-dashboard/OnboardedProduct';
+import { OnboardedProductResource } from '../../../../../api/generated/b4f-dashboard/OnboardedProductResource';
+import { Product, ProductInstitutionMap } from '../../../../../model/Product';
+
+type FilterConfig = {
+  institutionType: string;
+  categoryCode?: string;
+  allowedInstitutionTypes: ProductInstitutionMap;
+};
+
+const institutionTypeFilter = (
+  productsWithStatusActive: Array<Product>,
+  { institutionType, categoryCode, allowedInstitutionTypes }: FilterConfig
+): Array<Product> =>
+  productsWithStatusActive.filter((product) => {
+    // Check if product is allowed for the institution type
+    const allowedTypes = allowedInstitutionTypes[product.id] ?? allowedInstitutionTypes?.default;
+    if (!allowedTypes || !allowedTypes[institutionType]) {
+      return false;
+    }
+
+    // If categories are required, check if the institution's category is in the allowed list
+    if (allowedTypes[institutionType].categories) {
+      const categories = categoryCode && (allowedTypes[institutionType].categories as string);
+      return categories && categories.includes(categoryCode);
+    }
+
+    return true;
+  });
+
+const onboardingStatusFilter = (
+  productsWithStatusActive: Array<Product>,
+  onboardedProducts: Array<OnboardedProductResource>
+): Array<Product> => {
+  const onboardedProductIds = onboardedProducts.map((p) => p.productId);
+
+  return productsWithStatusActive.filter((product) => {
+    // For productsWithStatusActive with active base version, show eligible children
+    if (product.subProducts && product.subProducts?.length > 0) {
+      return product.subProducts.some(
+        (child) => !onboardedProductIds.includes(child.id ?? '') && child.status === 'ACTIVE'

```

Comment:
Instead of hardcoding strings like 'ACTIVE', consider defining them as constants or enums to avoid typos and enhance maintainability.

Example:

```
enum ProductStatus {
  ACTIVE = 'ACTIVE',
  // other statuses
}
```